cdap-spark-core2_2.11/src/test/resources/META-INF/services/io.cdap.cdap.app.runtime.spark.SparkProgramRuntimeProvider

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright © 2022 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+
+io.cdap.cdap.app.runtime.spark.Spark2ProgramRuntimeProvider
```

Comment:
In your test, you can just load with `ProgramRuntimeProvider.class` so that you don't need this service definition file?

Reply:
I think we need to have a specific service definition regardless because it is being used in unit tests which reads from test-classes

Reply:
Actually it seems like using the original ProgramRuntimeProvider worked like you mentioned, fixed.